warn-stranded.lua

```diff
@@ -0,0 +1,584 @@
+    local ignored = deserializeIgnoredUnits()
+
+    if groupNumber > #groups then
+        print('Group '..groupNumber..' does not exist')
+        return false
+    end
+
+    if groups[groupNumber]['mainGroup'] then
+        print('Group '..groupNumber..' is the main group of dwarves. Cannot toggle.')
+        return false
+    end
+
+    local group = groups[groupNumber]
+
+    local allIgnored = true
+    for _, unit in ipairs(group['units']) do
+        if not unitIgnored(unit, ignored) then
+            allIgnored = false
+            goto process
+        end
+    end
+    ::process::
+
+    for _, unit in ipairs(group['units']) do
+        local isIgnored = unitIgnored(unit, ignored)
+
+        if allIgnored == isIgnored then
+            toggleUnitIgnore(unit, ignored)
+        end
+    end
+
+    return true
+end
+
+-- ===============================================================
+--                   Graphical Interface
+-- ===============================================================
+warning = defclass(warning, gui.ZScreenModal)
+
+function warning:init(info)
+    self:addviews{
+        widgets.Window{
+            view_id = 'main',
+            frame={w=80, h=25},
+            min_size={w=60, h=25},
+            frame_title='Stranded Citizen Warning',
+            resizable=true,
+            autoarrange_subviews=true,
+            subviews = {
+                widgets.List{
+                    frame={h=15},
+                    view_id = 'list',
+                    text_pen = { fg = COLOR_GREY, bg = COLOR_BLACK },
+                    cursor_pen = { fg = COLOR_BLACK, bg = COLOR_GREEN },
+                    on_submit=self:callback('onIgnore'),
+                },
+                widgets.Panel{
+                    frame={h=5},
+                    autoarrange_subviews=true,
+                    subviews = {
+                        widgets.HotkeyLabel{
+                            key='SELECT',
+                            label='Toggle Ignore',
+                        },
+                        widgets.HotkeyLabel{
+                            key='CUSTOM_G',
+                            label='Toggle Group',
+                            on_activate = self:callback('onToggleGroup'),
+                        },
+                        widgets.HotkeyLabel{
+                            key = 'CUSTOM_SHIFT_I',
+                            label = 'Ignore All',
+                            on_activate = self:callback('onIgnoreAll'),
+                        },
+                        widgets.HotkeyLabel{
+                            key = 'CUSTOM_SHIFT_C',
+                            label = 'Clear All Ignored',
+                            on_activate = self:callback('onClear'),
+                        },
+                        widgets.HotkeyLabel{
+                            key = 'CUSTOM_Z',
+                            label = 'Zoom to unit',
+                            on_activate = self:callback('onZoom'),
```

Comment:
it seems to me that the player would *always* want to zoom to the selected unit. Could this be an `on_select` handler for the List instead?